contents/java/Spring.md

```diff
@@ -0,0 +1,193 @@
+  
+    <bean id="obj" class="com.company.Employee">  
+    <property name="id">  
+    <value>20</value>  
+    </property>  
+    <property name="name">  
+    <value>Amy</value>  
+    </property>  
+    </bean>  
+    </beans>
+```
+We are providing the information into the bean by this file. The property element invokes the setter method. The value subelement of property will assign the specified value.
+   
+To better understand how this dependency injection work, we can have a `Test.java` looks like the following:
+```java
+    public class Test {  
+        public static void main(String[] args) {   
+            Resource r=new ClassPathResource("applicationContext.xml");  
+            BeanFactory factory=new XmlBeanFactory(r);  
+          
+            Employee e=(Employee)factory.getBean("obj");  
+            s.display();
+        }  
+    }  
+```
+The out put of the above code will be `20 Amy`.
+
+### Spring JdbcTemplate
+
+Spring JdbcTemplate is a powerful mechanism to connect to the database and execute SQL queries. It internally uses
+JDBC API, but eliminates a lot of problems of it.
+    
+The problems of traditional JDBC API are listed as follows:
+- We need to write a lot of code before and after executing the query, such as creating connection,
+statement, closing resultset, connection etc.
+- We need to perform exception handling code on the database logic.
+- We need to handle transaction.
+- Repetition of all these codes from one to another database logic is a time consuming task.
+
+Spring JdbcTemplate eliminates all the above mentioned problems of JDBC API.
+It provides you methods to write the queries directly, so it saves a lot of work and time.
+![Spring JdbcTemplate](springJDBC.png "How much Spring saves you")
+
+### Spring AOP
+
+Aspect Oriented Programming (AOP) compliments OOPs in the sense that it provides modularity in aspect instead of class.
+AOP breaks the program logic into distinct parts called concerns. A cross-cutting concern is a concern that can affect
+the whole application and should be centralized in one location in code as much as possible. Examples of these concerns
+include transaction management, authentication, logging, security etc.
+
+By separating application business logic from system services, Spring Framework supports Aspect Oriented Programming
+and enables cohesive development. It provides the way to dynamically add the
+cross-cutting concern before, after or around the actual logic using simple
+pluggable configurations. It also allows easier to maintainance of the code. You can add/remove concerns
+without recompiling complete sourcecode simply by changing configuration file by applying aspects using XML configuration.
+
+## Why use Spring?
+
+Spring makes programming Java quicker, easier, and safer for everybody.
+Like all other general Java frameworks, Spring helps us focus on the core task rather than the boilerplate associated with it.
```

Comment:
'Like all other general Java frameworks' → 'Similar to other Java frameworks (e.g. Grails, Play)'